packages/extension-chakra-storefront/src/mocks/datacloud-mock-data.js

```diff
@@ -6,12 +6,29 @@
  */
 import {expect} from '@jest/globals'
 
+expect.extend({
+    toBeStringOrUndefined(received) {
+        const pass = typeof received === 'string' || received === undefined
+        if (pass) {
+            return {
+                message: () => `expected ${received} not to be a string or undefined`,
+                pass: true
+            }
+        } else {
+            return {
+                message: () => `expected ${received} to be a string or undefined`,
+                pass: false
+            }
+        }
+    }
+})
+
 export const mockLoginViewPageEvent = {
     events: [
         expect.objectContaining({
             guestId: 'guest-usid',
             siteId: 'RefArch',
-            sessionId: expect.any(String),
+            sessionId: expect.toBeStringOrUndefined(),
```

Comment:
Since we avoid mocking js-cookie, this line in the source code would return undefined: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/8f938014b51d51083bd7a46b66b1933436f61491/packages/extension-chakra-storefront/src/hooks/use-datacloud.js#L421

So in this file, we test whether `sessionId` is either string or undefined.

I think doing it this way still preserve the original intention of the test. We don't need to test whether the value is correct; we can safely assume that the third-party `Cookies.get(...)` works as expected. But we do need to make sure that the object contains the `sessionId` key.